Scripts/RemediationScripts/Remediate-SecureFTPDeploymentForAppServices.ps1

```diff
@@ -0,0 +1,1114 @@
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalAppServicesWithFTPStateAllAllowed)] out of [$($totalAppServices)] App Service(s) with FTP state not Configured." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    Write-Host $([Constants]::SingleDashLine)
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableSecureFTPDeploymentForAppServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    if (-not $DryRun)
+    {
+       
+        Write-Host "Backing up App Services details to [$($backupFolderPath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        $backupFile = "$($backupFolderPath)\AppServicesWithNonCompliantFTPDeployment.csv"
+        $appServicesWithFTPStateAllAllowed | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "App Services details have been backed up to [$($backupFile)]." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+        
+        Write-Host "FTP state will be updated on the production slot and all non-production slots for all App Services." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+
+        $userInputforFTPState = @()
+        if (-not $Force)
+            {
+                if($FTPState)
+                {
+                    Write-Host "FTP State is already provided as input parameter. " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+                }
+                else
+                {
+                Write-Host "Do you want to upadate FTP State? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+                $userInput = Read-Host -Prompt "(Y|N)"
+                Write-Host $([Constants]::SingleDashLine)
+                if($userInput -eq "Y")
+                {
+                    #Write-Host "User has provided consent to secure FTP Deployments on the production slot and all non-production slots for all App Services." -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                    Write-Host "Please select 1 to set FTP State as FTPSOnly or select 2 to set FTP State Disabled on the production slot and all non-production slots for all App Services" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+                    $userInput = Read-Host -Prompt "(1|2)"
```

Comment:
Always good to use do while loop here, with option to exist loop. So that if by mistake user selects wroing input say 3, all above steps he/she will need to run/wait again.
Let's keep a note of this and do this enhancement along with auto remediation support.